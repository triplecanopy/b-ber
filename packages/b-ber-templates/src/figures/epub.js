import { media, iframe, figure, unsupported } from './helpers'

const epub = {
  portrait: data => figure({ data, ratioName: 'portrait' }),
  landscape: data => figure({ data, ratioName: 'landscape' }),
  'portrait-high': data => figure({ data, ratioName: 'portrait-high' }),
  square: data => figure({ data, ratioName: 'square' }),
  audio: data => media({ ...data, applyInlineClasses: true }),
  video: data => media({ ...data, applyInlineClasses: true }),
  iframe: data => iframe({ ...data, applyInlineClasses: true }),
  vimeo: data => unsupported({ data, applyInlineClasses: true }),
  soundcloud: data => unsupported({ data, applyInlineClasses: true }),
}

export default epub
