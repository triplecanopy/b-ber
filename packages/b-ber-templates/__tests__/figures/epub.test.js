import fs from 'fs-extra'
import epub from '../../src/figures/epub'

jest.mock('@canopycanopycanopy/b-ber-lib/State', () => ({}))

afterAll(() => fs.remove('_project'))

describe('templates.epub', () => {
  test('creates the markup', () => {
    const data = {
      inline: true,
      classes: 'test',
      id: 1,
      ref: 1,
      alt: 'test',
      source: 'test',
      caption: 'test',
      attrString: 'autoplay="false"',
      sourceElements: '<source />',
      mediaType: 'audio-or-video',
      poster: 'test.jpg',
      title: 'test-title',
      width: 'test-width',
      height: 'test-height',
    }

    const dataFiguresPage = Object.assign({}, data, { inline: false })
    const dataPagebreakBefore = Object.assign({}, data, {
      classes: 'break-before',
    })

    expect(epub.portrait(data)).toMatchSnapshot()
    expect(epub.landscape(data)).toMatchSnapshot()
    expect(epub['portrait-high'](data)).toMatchSnapshot()
    expect(epub.square(data)).toMatchSnapshot()
    expect(epub.audio(data)).toMatchSnapshot()
    expect(epub.video(data)).toMatchSnapshot()
    expect(epub.iframe(data)).toMatchSnapshot()

    expect(epub.portrait(dataFiguresPage)).toMatchSnapshot()
    expect(epub.landscape(dataFiguresPage)).toMatchSnapshot()
    expect(epub['portrait-high'](dataFiguresPage)).toMatchSnapshot()
    expect(epub.square(dataFiguresPage)).toMatchSnapshot()
    expect(epub.audio(dataFiguresPage)).toMatchSnapshot()
    expect(epub.video(dataFiguresPage)).toMatchSnapshot()
    expect(epub.iframe(dataFiguresPage)).toMatchSnapshot()

    expect(epub.portrait(dataPagebreakBefore)).toMatchSnapshot()
    expect(epub.landscape(dataPagebreakBefore)).toMatchSnapshot()
    expect(epub['portrait-high'](dataPagebreakBefore)).toMatchSnapshot()
    expect(epub.square(dataPagebreakBefore)).toMatchSnapshot()
    expect(epub.audio(dataPagebreakBefore)).toMatchSnapshot()
    expect(epub.video(dataPagebreakBefore)).toMatchSnapshot()
    expect(epub.iframe(dataPagebreakBefore)).toMatchSnapshot()
  })
})
