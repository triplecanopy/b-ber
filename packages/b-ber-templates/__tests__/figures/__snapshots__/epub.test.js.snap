// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates.epub creates the markup 1`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.epub creates the markup 2`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--landscape test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.epub creates the markup 3`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait-high test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.epub creates the markup 4`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.epub creates the markup 5`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.epub creates the markup 6`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.epub creates the markup 7`] = `
"
    
      <div class=\\"figure__large figure__large--iframe\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"iframe\\">
              <iframe src=\\"test\\" webkitallowfullscreen=\\"webkitallowfullscreen\\" mozallowfullscreen=\\"mozallowfullscreen\\" allowfullscreen=\\"allowfullscreen\\" frameborder=\\"0\\" />
            </div>

            <div class=\\"figcaption\\">
                <p class=\\"small\\">
                  test<br/>
                  <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.epub creates the markup 8`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--portrait\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.epub creates the markup 9`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--landscape\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.epub creates the markup 10`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--portrait-high\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.epub creates the markup 11`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--square\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    <a href=\\"1.xhtml#ref1\\">
      <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\"/>
    </a>
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    </section>
"
`;

exports[`templates.epub creates the markup 12`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    </section>
    "
`;

exports[`templates.epub creates the markup 13`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--media figure__inline--square test\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    </section>
    "
`;

exports[`templates.epub creates the markup 14`] = `
"
    <section epub:type=\\"loi\\" class=\\"chapter figures\\">
      <div class=\\"figure__large figure__large--iframe\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"iframe\\">
              <iframe src=\\"test\\" webkitallowfullscreen=\\"webkitallowfullscreen\\" mozallowfullscreen=\\"mozallowfullscreen\\" allowfullscreen=\\"allowfullscreen\\" frameborder=\\"0\\" />
            </div>

            <div class=\\"figcaption\\">
                <p class=\\"small\\">
                  test<br/>
                  <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    </section>
    "
`;

exports[`templates.epub creates the markup 15`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"portrait\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.epub creates the markup 16`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--landscape break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"landscape\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.epub creates the markup 17`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--portrait-high break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"portrait-high\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.epub creates the markup 18`] = `
"
    
      <div class=\\"figure__large figure__inline figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            
    
      <img class=\\"square\\" alt=\\"test\\" src=\\"../images/test\\"/>
    
    <div class=\\"figcaption\\">
      <p class=\\"small\\">test</p>
    </div>
    
          </div>
        </figure>
      </div>
    
"
`;

exports[`templates.epub creates the markup 19`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.epub creates the markup 20`] = `
"
    
      <div class=\\"figure__large figure__large--media figure__inline--square break-before\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"audio-or-video\\">
              <audio-or-video autoplay=\\"false\\">
                <source />
                <div class=\\"media__fallback__audio-or-video media__fallback--image\\">
                  <figure>
                    <img src=\\"test.jpg\\" alt=\\"Media fallback image\\"/>
                  </figure>
                </div>
                <p class=\\"media__fallback__audio-or-video media__fallback--text\\">Your device does not support the HTML5 audio-or-video API.</p>
              </audio-or-video>
            </div>
            <div class=\\"figcaption\\">
              <p class=\\"small\\">
                test<br/>
                <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;

exports[`templates.epub creates the markup 21`] = `
"
    
      <div class=\\"figure__large figure__large--iframe\\">
        <figure id=\\"1\\">
          <div class=\\"figure__items\\">
            <div class=\\"iframe\\">
              <iframe src=\\"test\\" webkitallowfullscreen=\\"webkitallowfullscreen\\" mozallowfullscreen=\\"mozallowfullscreen\\" allowfullscreen=\\"allowfullscreen\\" frameborder=\\"0\\" />
            </div>

            <div class=\\"figcaption\\">
                <p class=\\"small\\">
                  test<br/>
                  <a href=\\"1.xhtml#ref1\\">Return</a>
              </p>
            </div>
          </div>
        </figure>
      </div>
    
    "
`;
